h5py/_hl/compat.py

```diff
@@ -3,8 +3,9 @@
 """
 import sys
 from os import fspath, fsencode, fsdecode
+from ..version import hdf5_built_version_tuple
 
-WINDOWS_ENCODING = "mbcs"
+WINDOWS_ENCODING = "utf-8" if hdf5_built_version_tuple >= (1, 10, 6) else "mbcs"
```

Comment:
Where did you find that this is the case? I've found [the blog post announcing 1.10.6](https://www.hdfgroup.org/2019/12/release-of-hdf5-1-10-6-newsletter-170/) and the [release notes](https://confluence.hdfgroup.org/display/support/HDF5%201.10.6#releasenotes), neither of which seem to mention it. I also found [forum](https://forum.hdfgroup.org/t/non-english-characters-in-hdf5-file-name/4627) [threads](https://forum.hdfgroup.org/t/wide-string-equivalent-of-h5fcreate-or-h5fopen/5482) about the issue, but no-one saying it's been fixed. :confused:

Reply:
#839 has comments about the fix arriving. Interestingly https://jira.hdfgroup.org/browse/HDFFV-11026?jql=labels%20%3D%20Windows suggests it wasn't announced due to problems with testing (it's not clear which encoding the reporter used).

Reply:
@epourmal do you have any newer information about unicode filenames on Windows? There are some hints that it was done for 1.10.6, while [HDFFV-10691](https://jira.hdfgroup.org/browse/HDFFV-10691) is 'in progress' and seems to say it might be done for 1.12.1.

If this is now supported, do we pass UTF-8 filenames to `H5Fopen`, or do we need to do something special for it?